### PR TITLE
[API-1372]: Improvements based on customer feedback

### PR DESCRIPTION
**Pull request by therockstorm**
## Summary

- Parse properties for suppliedRevisionId
- Add spinner for long running tasks
- Handle new errors from api-client
- Add flag to configure parallelism

## Possible Regressions

n/a

## Dependencies

n/a

### CODE REVIEW

File: src/create-template/pvs.ts [hunk 1]
```diff
@@ -78,8 +91,31 @@ export const processPvs = (
...
+          if (pId && iId) partIdToIterationId[pId.value] = iId.value;
+        }
+      })
+    );
+
+  const getRevisionId = (name: string): string => {
+    for (const k of Object.keys(partIdToIterationId)) {
+      if (k.includes(name)) return partIdToIterationId[k];
+    }
+    return DefaultPartRevision;
```
Comment (therockstorm) [inline src/create-template/pvs.ts:116]: This works the same if these two properties don't exist so is safe to publish, but will want to wait to ensure these are the correct ones.

File: src/commands/render-image.ts [hunk 2]
```diff
@@ -47,20 +47,25 @@ Image written to 'f79d4760-0b71-44e4-ad0b-22743fdd4ca3.jpeg'.
...
+      const renderArgs = {
+        client: await VertexClient.build({ basePath: flags.basePath }),
+        renderReq: { id: args.id, height: flags.height, width: flags.width },
+      };
+      const renderRes = await (flags.resource === 'scene'
+        ? renderScene(renderArgs)
+        : renderSceneView(renderArgs));
+      if (parseInt(renderRes.headers['content-length'], 10) < 140) {
+        this.error(`Received empty image for ${flags.resource} ${args.id}.`);
+      }
```
Comment (danschultz) [inline src/commands/render-image.ts:60]: This seems like an area in our API that may need another look? There's valid reasons for an empty image (no items, or all the scene items are hidden). Maybe there's some room for improvement at looking at the status codes on the generated image and translating those into HTTP status codes.

File: src/create-template/pvs.ts [hunk 3]
```diff
@@ -78,8 +91,31 @@ export const processPvs = (
   verbose: boolean,
   root?: string
 ): ExtendedTemplateItem[] => {
+  const partIdToIterationId: { [k: string]: string } = {};
```
Comment (danschultz) [inline src/create-template/pvs.ts:94]: `const partIdToIterationId: Record<string, string> = {};`

**Review by danschultz — APPROVED** [review]

File: src/commands/render-image.ts (hunk 2 above)
Comment (therockstorm) [reply src/commands/render-image.ts:60]: Made a note of this, I'll get a ticket created.

File: src/create-template/pvs.ts (hunk 3 above)
Comment (therockstorm) [reply src/create-template/pvs.ts]: Updated

**Review by jareddellitt — APPROVED** [review]

File: src/create-template/pvs.ts [hunk 4]
```diff
@@ -120,21 +141,59 @@ export const processPvs = (
...
+    transform: !t || is4x4Identity(t) ? undefined : toTransform(t),
+  };
+}
 
-  recurse(components, components[rootIndex(components, root)]);
+function getRevisionId(idx: number, properties?: Properties): string {
+  if (!properties) return DefaultPartRevision;
+
+  for (const sectionProp of properties.properties) {
+    const propCompRef = sectionProp.property_component_ref[idx];
```
Comment (therockstorm) [inline src/create-template/pvs.ts:184]: These use the same indexes as the `section_structure` part of the file. This is much nicer than the string compare I was doing to find the corresponding properties.

**Pull request merged**